src/Raven.Server/Documents/ETL/Providers/Queue/Kafka/KafkaEtl.cs

```diff
@@ -135,25 +162,171 @@ void ReportHandler(DeliveryReport<string, byte[]> report)
             }
-            catch (Exception e)
+
+            var msgBuilder = new StringBuilder($"ETL process: {Name}. Failed to load data for the producer instance. " +
+                                               $"Please check your Kafka security configuration and ensure all of the (RESOURCE : OPERATION) ACL bindings listed below are set to ALLOW: {Environment.NewLine}");
+
+            if ((validTransactionDescribeAcl && validTransactionWriteAcl) == false)
             {
-                if (Logger.IsOperationsEnabled)
-                    Logger.Operations($" ETL process: {Name}. Aborting Kafka transaction failed.", e);
+                msgBuilder.Append($"Transactional ID '{TransactionalId}' : ");
+                if (validTransactionWriteAcl == false)
+                    msgBuilder.Append("WRITE operation ");
+                if (validTransactionDescribeAcl == false)
+                    msgBuilder.Append("DESCRIBE operation");
+                msgBuilder.Append(Environment.NewLine);
             }
 
-            throw new QueueLoadException(ex.Message, ex);
+            
+            if (validTopicWriteAcl == false)
+                msgBuilder.Append($"Topic '{topicName}' : WRITE operation");
+
+            message = msgBuilder.ToString();
+
+        }
+        catch (Exception)
+        {
+            return false;
         }
 
-        return count;
+        return true;
+    }
+
+    private static bool AssertAclMatchingName(string expectedName, AclBinding aclBinding)
+    {
+        return aclBinding.Pattern.ResourcePatternType switch
+        {
+            ResourcePatternType.Literal => aclBinding.Pattern.Name == expectedName,
+            ResourcePatternType.Prefixed => expectedName.StartsWith(aclBinding.Pattern.Name),
+            _ => throw new NotSupportedException()
+        };
+    }
+    
+
+    private void ThrowTransactionalIdAclAuthorizationException(Exception e)
+    {
+        string msg = $"ETL process: {Name}. Failed to initialize transactions for the producer instance. " +
+                     $"Add an required ACL permissions to your API Key that will make WRITE and DESCRIBE operations possible for transactional ID '{TransactionalId}'. ";
+                                    
+        if (Logger.IsOperationsEnabled)
+            Logger.Operations(msg, e);
+        // no need to abort transaction yet, this method is called when _producer.InitTransactions() fails
+        throw new QueueLoadException(msg, e);
+    }
+
+    private void ThrowTopicAclAuthorizationException(Exception e, string currentTopicName)
+    {
+        var errorMessageBuilder = new StringBuilder($"ETL process: {Name}. Failed to initialize transactions for the producer instance.{Environment.NewLine}" +
+                                                    $"Add an required ACL permissions to your API Key that will make WRITE operations on the '{currentTopicName}' topic possible.");
+        AbortTransactionAfterException(errorMessageBuilder);
```

Comment:
I don't think it's a good idea to call it here. It's not expected that a method going to throw will also do some extra logic

Reply:
I'm renaming the method name to `AbortKafkaTxAndThrowTopiAclAuthorizationException`- it'll now be more meaningful, its main purpose is to abort the transaction (safe exit before throwing)

Reply:
update: after introducing `TryAbortTransactionAfterException` got rid of this method